test/output_mappings.tsv

```diff
@@ -0,0 +1,2058 @@
+1:1526453:C:G	1	ENSG00000197785	ATAD3A	splice_region_variant	0
+1:15440344:C:T	1	ENSG00000162438	CTRC	stop_gained	0
+1:154572104:G:A	1	ENSG00000160716	CHRNB2	synonymous_variant	0
+1:154584092:C:T	1	ENSG00000160710	ADAR	3_prime_UTR_variant	0
+1:155904493:T:G	1	ENSG00000143622	RIT1	missense_variant	0
+1:156158061:C:T	1	ENSG00000196189	SEMA4A	intron_variant	0
+1:156852093:A:G	1	ENSG00000027644	INSRR	missense_variant	0
+11:57599962:C:T	1	ENSG00000149131	SERPING1	synonymous_variant	0
+1:158657627:A:G	1	ENSG00000163554	SPTA1	synonymous_variant	0
+1:159045460:C:T	1	ENSG00000163565	IFI16	missense_variant	0
+11:59855881:C:T	1	ENSG00000134827	TCN1	missense_variant	0
+1:160042259:C:A	1	ENSG00000177807	KCNJ10	missense_variant	0
+1:161306854:C:T	1	ENSG00000158887	MPZ	stop_gained	0
+1:161306855:A:G	1	ENSG00000158887	MPZ	missense_variant	0
+1:161328480:C:A	1	ENSG00000143252	SDHC	synonymous_variant	0
+11:61392608:C:G	1	ENSG00000187049	TMEM216	5_prime_UTR_variant	0
+11:61397887:C:G	1	ENSG00000187049	TMEM216	missense_variant	0
+11:62613504:G:A	1	ENSG00000149489	ROM1	missense_variant	0
+11:62631154:C:A	1	ENSG00000089597	GANAB	synonymous_variant	0
+11:62690468:G:A	1	ENSG00000168000	BSCL2	missense_variant	0
+11:62690799:G:T	1	ENSG00000168000	BSCL2	missense_variant	0
+11:62702345:C:T	1	ENSG00000168000	BSCL2	intron_variant	0
+11:63909090:G:A	1	ENSG00000072518	MARK2	synonymous_variant	0
+11:6393629:G:A	1	ENSG00000166311	SMPD1	missense_variant	0
+11:64753640:G:A	1	ENSG00000068976	PYGM	missense_variant	0
+11:64804588:G:A	1	ENSG00000133895	MEN1	stop_gained	0
+11:64807554:G:A	1	ENSG00000133895	MEN1	stop_gained	0
+11:65718189:C:T	1	ENSG00000172922	RNASEH2C	3_prime_UTR_variant	0
+11:65871357:T:C	1	ENSG00000172638	EFEMP2	missense_variant	0
+11:6604283:T:A	1	ENSG00000166333	ILK	synonymous_variant	0
+11:6615256:C:T	1	ENSG00000166340	TPP1	missense_variant	0
+11:6616431:A:C	1	ENSG00000166340	TPP1	missense_variant	0
+11:67609539:G:T	1	ENSG00000167792	NDUFV1	synonymous_variant	0
+1:168314237:G:A	1	ENSG00000143178	TBX19	3_prime_UTR_variant	0
+11:68760238:C:T	1	ENSG00000110090	CPT1A	missense_variant	0
+11:68784813:A:G	1	ENSG00000110090	CPT1A	splice_donor_variant	0
+11:68917823:G:A	1	ENSG00000132740	IGHMBP2	missense_variant	0
+11:68929230:G:A	1	ENSG00000132740	IGHMBP2	missense_variant	0
+1:169524916:A:G	1	ENSG00000198734	F5	splice_region_variant	0
+1:17027759:C:T	1	ENSG00000117118	SDHB	missense_variant	0
+11:71444115:C:T	1	ENSG00000172893	DHCR7	missense_variant	0
+1:171652368:G:A	1	ENSG00000034971	MYOC	missense_variant	0
+11:72106204:G:A	1	ENSG00000184154	LRTOMT	missense_variant	0
+11:72106204:G:A	1	ENSG00000284844	AP000812.4	missense_variant	0
+11:72196097:G:A	1	ENSG00000110195	FOLR1	missense_variant	0
+11:72330077:G:A	1	ENSG00000162129	CLPB	intron_variant	0
+11:72358891:A:G	1	ENSG00000162129	CLPB	missense_variant	0
+1:172659066:G:A	1	ENSG00000117560	FASLG	upstream_gene_variant	37
+1:172659066:G:A	1	ENSG00000117560	FASLG	upstream_gene_variant	78
```

Comment:
I was curious why the output is longer than the input.

Is the last number the distance in nucleotides to the gene? is it ok if there are two identical lines that only change the distance (like this and the previous one)? is it possible that the variant affects the same gene (FASLG) in 2 different locations?

I also see the variant at line 334 affects 2 different genes, which I guess is less strange if the genes overlap ??
```
1:225922234:C:T 1       ENSG00000143811 PYCR2   synonymous_variant      0
1:225922234:C:T 1       ENSG00000255835 AL117348.2      synonymous_variant      0
```
I have the list of extra variants but haven't checked them all

Reply:
In general it is expected that a variant can produce more than one gene+consequence mapping, because a variant can indeed affect multiple gene. The second case you've pointed out is exactly this: position 1:225922234 overlaps two genes; one is “normal” and another one appears to be some sort of the readthrough transcript uniting two nearby genes. Here's the situation in the genome, with the variant highlighted with yellow on the right side:

![hgt_genome_699fb_d78110](https://user-images.githubusercontent.com/10669118/74861248-ee0c8900-5341-11ea-8fa5-407f56f2a36a.png)

In this case it is perfectly normal to have two output records for one input variant.

Reply:
However, the first case which you described is another thing and will need adjustments. This is a gene with two transcripts starting at slightly different positions, and a variant is affecting them (the transcripts) with the same consequence, but at slightly different distances. In this case we don't actually want two records because the distance to the transcript is not taken into account in the downstream processing, so it will just lead to duplicate evidence strings. This is not a big problem (duplicate evidence strings are filtered out by OT ingest pipeline), but still something which we should fix.

Outputting distances for upstream and downstream variants is actually a part of my proposal to unify/simplify their handling which was eventually not accepted by OT (https://github.com/EBIvariation/vep-mapping-pipeline/pull/1 was concerned by disabling part of that functionality by default). I guess the simplest thing I can do here is to not report distance by default (as was done by the old pipeline), and then we won't get several records because they will be deduplicated prior to outputting. I think I'll implement that.

Reply:
@jmmut Distance calculation disabled in https://github.com/EBIvariation/vep-mapping-pipeline/pull/2/commits/821cc362e462b101bcefd513c6a2fd76fb7bb760, test data updated in https://github.com/EBIvariation/vep-mapping-pipeline/pull/2/commits/cad3a75a34d1101ed9b11d78f9542eaa9e7ab835 — in that second commit you can see that the duplicates have disappeared